aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -431,7 +431,10 @@ Tensor linalg_matrix_power_impl(
       // Clone input to include result in the autograd graph
       out = self.clone(at::MemoryFormat::Contiguous);
     }
-    return out.copy_(at::eye(self.size(-2), self.options()));
```

Comment:
Is this change necessary? It's fine to keep it, but I'd like to know whether `at::eye(self.size(-2), self.options())` still works and correct defaults are chosen.

Reply:
Unfortunately this is required (and is a BC breaking change). It isn't possible to disambiguate between `int, opt<int>, opt<int>, options` and any other combination. So the C++ API changes because of this PR. The Python usage is the same though.

Reply:
> @HaoZeke - This PR is currently marked as a draft. Is it ready for review?

Almost, will take it out of draft status for more discussion after resolving the remaining comments.

Reply:
Changed to reflect a more BC respecting implementation as suggested :)